src/System.Data.SqlClient/src/System.Data.SqlClient.csproj

```diff
@@ -11,16 +11,19 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <PackageTargetFramework>dotnet5.5</PackageTargetFramework>
     <UsePackageTargetRuntimeDefaults>true</UsePackageTargetRuntimeDefaults>
+    <ProjectGuid>{D4550556-4745-457F-BA8F-3EBF3836D6B4}</ProjectGuid>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(TargetsWindows)' != 'true' ">
     <DefineConstants>$(DefineConstants);MANAGED_SNI</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(TargetsWindows)' == 'true' ">
```

Comment:
For this, please use a different CS file to compiler on non-windows platform. 
Let's not define a different Constant. 
Wrap the existing Interop.cs in a windows only compilation group.